src/inmanta/data/__init__.py

```diff
@@ -4483,6 +4483,38 @@ async def trim(cls, environment: UUID, connection: Optional[Connection] = None)
             connection=connection,
         )
 
+    @classmethod
+    async def get_aggregated_deployment_status(
+        cls, environment_id: UUID, resource_ids: list[ResourceIdStr]
+    ) -> dict[str, dict[const.NonDeployingResourceState, int]]:
```

Comment:
We should really prevent these kinds of datastructures with deeply nested dicts. This makes the code so much harder to read. Why not use an object instead?